File: lang-anteriores/compare_langs.js
const fs = require('fs');

function parseLangFile(content) {
    // Extract the $lang array content
    const langMatch = content.match(/\$lang = \[(.*?)\];/s);
    if (!langMatch) {
        throw new Error("Could not find $lang array");
    }
    
    const langContent = langMatch[1];
    
    // Parse the key-value pairs using regex
    const pairs = [];
    const regex = /'([^']*)'\s*=>\s*'([^']*)'/g;
    let match;
    
    while ((match = regex.exec(langContent)) !== null) {
        const key = match[1];
        const value = match[2];
        pairs.push({ key, value });
    }
    
    // Convert to object
    const langObj = {};
    for (const pair of pairs) {
        langObj[pair.key] = pair.value;
    }
    
    return langObj;
}

function compareLangs(esFile, elFile) {
    const esContent = fs.readFileSync(esFile, 'utf-8');
    const elContent = fs.readFileSync(elFile, 'utf-8');
    
    const esLang = parseLangFile(esContent);
    const elLang = parseLangFile(elContent);
    
    // Find entries that exist in es_AR but not in el_GR
    const faltantes = {};
    for (const [key, value] of Object.entries(esLang)) {
        if (!elLang.hasOwnProperty(key)) {
            faltantes[key] = value;
        }
    }
    
    // Find entries that have different values between the two files
    const diferentes = {};
    for (const [key, esValue] of Object.entries(esLang)) {
        if (elLang.hasOwnProperty(key) && elLang[key] !== esValue) {
            diferentes[key] = {
                'es_AR': esValue,
                'el_GR': elLang[key]
            };
        }
    }
    
    return { faltantes, diferentes };
}

function main() {
    const { faltantes, diferentes } = compareLangs('es_AR.php', 'el_GR.php');
    
    console.log("=== ENTRIES MISSING IN el_GR.php ===\n");
    for (const [key, value] of Object.entries(faltantes)) {
        console.log(`'${key}' => '${value}',`);
    }
    
    console.log("\n=== ENTRIES WITH DIFFERENT VALUES ===\n");
    for (const [key, values] of Object.entries(diferentes)) {
        console.log(`'${key}' => es_AR: '${values['es_AR']}', el_GR: '${values['el_GR']}',`);
    }
}

main();

File: lang-anteriores/compare_langs.py
import re
import os

def load_lang_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Extract the $lang array using regex
    pattern = r'\$lang = \[(.*?)\];'
    matches = re.search(pattern, content, re.DOTALL)
    
    if not matches:
        raise ValueError(f"Could not find $lang array in {filepath}")
    
    lang_content = matches.group(1)
    
    # Parse the key-value pairs
    lang_dict = {}
    
    # Match 'key' => 'value' patterns
    pair_pattern = r"'([^']*)'\s*=>\s*'([^']*)'"
    matches = re.findall(pair_pattern, lang_content)
    
    for key, value in matches:
        lang_dict[key] = value
    
    return lang_dict

def compare_langs(es_file, el_file):
    es_lang = load_lang_file(es_file)
    el_lang = load_lang_file(el_file)
    
    # Find entries that exist in es_AR but not in el_GR
    faltantes = {}
    for key, value in es_lang.items():
        if key not in el_lang:
            faltantes[key] = value
    
    # Find entries that have different values between the two files
    diferentes = {}
    for key, es_value in es_lang.items():
        if key in el_lang and el_lang[key] != es_value:
            diferentes[key] = {
                'es_AR': es_value,
                'el_GR': el_lang[key]
            }
    
    return faltantes, diferentes

def main():
    es_file = 'es_AR.php'
    el_file = 'el_GR.php'
    
    faltantes, diferentes = compare_langs(es_file, el_file)
    
    print("=== ENTRIES MISSING IN el_GR.php ===\n")
    for key, value in faltantes.items():
        print(f"'{key}' => '{value}',")
    
    print("\n=== ENTRIES WITH DIFFERENT VALUES ===\n")
    for key, values in diferentes.items():
        print(f"'{key}' => es_AR: '{values['es_AR']}', el_GR: '{values['el_GR']}',")

if __name__ == "__main__":
    main()